cpp/tests/utilities/cudf_test_utils.cuh

```diff
@@ -36,6 +38,38 @@
 using gdf_col_pointer = typename std::unique_ptr<gdf_column, 
                                                  std::function<void(gdf_column*)>>;
 
+
+// Prints a vector and valids
+template <typename T>
+void print_vector_and_valid(T * v,
+                        gdf_valid_type * valid,
+                        const size_t num_rows)
+{
+  auto functor = [&valid, &v](int index) -> std::string {
+    if (gdf_is_valid(valid, index))
+      return std::to_string((int)v[index]);
+    return std::string("@");
+  };
+  std::vector<int> indexes(num_rows);
+  std::iota(std::begin(indexes), std::end(indexes), 0);
+  std::transform(indexes.begin(), indexes.end(), std::ostream_iterator<std::string>(std::cout, ", "), functor);
+  std::cout << std::endl;
+}
+
+static constexpr int ValidSize = 32;
+using ValidType = uint32_t;
+
+static size_t  valid_size(size_t column_length)
+{
+  const size_t n_ints = (column_length / ValidSize) + ((column_length % ValidSize) ? 1 : 0);
+  return n_ints * sizeof(ValidType);
+}
+
+static bool get_bit(const gdf_valid_type* const bits, size_t i)
+{
+  return  bits == nullptr? true :  bits[i >> size_t(3)] & (1 << (i & size_t(7)));
```

Comment:
FYI, there's no need to do the bit shifting manually.

The following code will be just as performant because the compiler will do the bit shift optimization for you. It has the benefit of being more readable as well.

`bits[i / 8] & (i % 8)`

Reply:
Removed code, using now the gdf_is_valid from gdf/utils.h.